### DISCUSSION TITLE
Update python versions in tests

### DISCUSSION THREAD

**Discussion by Tardo** [Tecnativa/doodba-copier-template#393] [General]
Time to drop python 3.7 and add 3.11 in the tests?
See: https://devguide.python.org/versions/

**Comment by pedrobaeza** [comment]
Well, are they specifically used in any Doodba Odoo version? If not, they can be simply removed.

**Comment by pleirb** [comment]
How's everybody? Does it have something to do with not being able to install on Ubuntu 22.04 as the manual suggests?
I can install all dependencies except:
**pipx install docker-compose**
```
Fatal error from pip prevented installation. Full pip output in file:
    /home/user/.local/pipx/logs/cmd_2023-07-26_21.58.23_pip_errors.log
pip seemed to fail to build package:
    PyYAML<6,>=3.10
Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    AttributeError: cython_sources
Error installing docker-compose.
```
**pipx install ensurepath**
```
Fatal error from pip prevented installation. Full pip output in file:
    /home/user/.local/pipx/logs/cmd_2023-07-26_22.02.09_pip_errors.log
Some possibly relevant errors from pip install:
    ERROR: Could not find a version that satisfies the requirement ensurepath (from versions: none)
    ERROR: No matching distribution found for ensurepath
Error installing ensurepath.
```

**Reply by pleirb** [reply]
At the moment I have solved installing through these steps:
1) Installing python3.7 from repositories
2) `python3.7 -m pip install --user pipx`
3) Editing `~/.bashrc` to make the environment variable permanent: `export PIPX_DEFAULT_PYTHON=/usr/bin/python3.7`
4) Installing of copier: 
```
pipx install copier==7.1.0
pipx inject copier 'pydantic<2'
```

**Comment by Tardo** [comment]
@pleirb On our side, we have made a fork and updated the dependencies. It is **not 100% tested** but so far it works without problems in our use case.
You can install it with: `pipx install https://github.com/admingrupoisonores/compose/archive/refs/heads/1.29.2f.zip`

**Reply by pleirb** [reply]
Thank you! I haven't gotten around to trying the branch you suggest yet, perhaps in future projects. Also, it wasn't just compose, I think ensurepath and something else that was giving me trouble. Will it be too hasty to think of modifying the installation steps in the manual for new users?
